### PR TITLE
mongodocstore: use two Open functions instead of options

### PR DESCRIPTION
**Pull request by jba**
Also, support id field URL parameter.

### CODE REVIEW

File: internal/docstore/mongodocstore/mongo.go [hunk 1]
```diff
@@ -146,30 +146,24 @@ type collection struct {
...
-
-// OpenCollection opens a MongoDB collection for use with Docstore.
-func OpenCollection(mcoll *mongo.Collection, opts Options) *docstore.Collection {
-	return docstore.NewCollection(newCollection(mcoll, opts))
+// OpenCollectionFunc opens a MongoDB collection for use with Docstore.
+// The idFunc argument is function that accepts a document and returns the value to
+// be used for the document ID (MongoDB's _id field). IDFunc should return nil if the
+// document is missing the information to construct an ID. This will cause all
+// actions, even Create, to fail.
+func OpenCollectionFunc(mcoll *mongo.Collection, idFunc func(docstore.Document) interface{}) *docstore.Collection {
```
Comment (shantuo) [inline internal/docstore/mongodocstore/mongo.go]: I think we should keep the options for future extensibility.

File: internal/docstore/mongodocstore/mongo.go [hunk 2]
```diff
@@ -146,30 +146,24 @@ type collection struct {
...
-	IDField string
-
-	// A function that accepts a document and returns the value to be used for the
-	// document ID (MongoDB's _id field).
-	// IDFunc should return nil if the document is missing the information to
-	// construct an ID. This will cause all actions, even Create, to fail.
-	IDFunc func(docstore.Document) interface{}
+// OpenCollection opens a MongoDB collection for use with Docstore.
+// The idField argument is the name of the document field to use for the document ID
+// (MongoDB's _id field).
```
Comment (shantuo) [inline internal/docstore/mongodocstore/mongo.go]: Add a comment that empty `idField` means using default "_id" name.

File: internal/docstore/mongodocstore/mongo_test.go [hunk 3]
```diff
@@ -139,6 +139,8 @@ func TestOpenCollection(t *testing.T) {
 		{"mongo:///mycollection", true},
 		// Missing collection name.
 		{"mongo://mydb/", true},
+		// valid parameter
```
Comment (shantuo) [inline internal/docstore/mongodocstore/mongo_test.go]:
```suggestion
		// Passing ID field path.
```

File: internal/docstore/mongodocstore/mongo.go (hunk 1 above)
Comment (vangent) [reply internal/docstore/mongodocstore/mongo.go]: +1, leave the Options even if they are empty.

Comment (jba) [reply internal/docstore/mongodocstore/mongo.go]: Done.

File: internal/docstore/mongodocstore/mongo.go (hunk 2 above)
Comment (jba) [reply internal/docstore/mongodocstore/mongo.go]: Done.

File: internal/docstore/mongodocstore/mongo_test.go (hunk 3 above)
Comment (jba) [reply internal/docstore/mongodocstore/mongo_test.go]: Done.

**Comment by jba** [comment]
PTAL

File: internal/docstore/mongodocstore/mongo.go [hunk 4]
```diff
@@ -118,14 +118,14 @@ type URLOpener struct {
 	// A Client is a MongoDB client that performs operations on the db, must be
 	// non-nil.
 	Client *mongo.Client
-
-	// Options specifies the options to pass to OpenCollection.
```
Comment (vangent) [inline internal/docstore/mongodocstore/mongo.go]: Restore options...?

Comment (jba) [reply internal/docstore/mongodocstore/mongo.go]: Done.

**Comment by jba** [comment]
PTAL.

File: internal/docstore/mongodocstore/mongo.go [hunk 5]
```diff
@@ -146,30 +149,26 @@ type collection struct {
...
 // OpenCollection opens a MongoDB collection for use with Docstore.
-func OpenCollection(mcoll *mongo.Collection, opts Options) *docstore.Collection {
-	return docstore.NewCollection(newCollection(mcoll, opts))
+// The idField argument is the name of the document field to use for the document ID
+// (MongoDB's _id field). If it is empty, the field "_id" will be used.
+func OpenCollection(mcoll *mongo.Collection, idField string, _ *Options) *docstore.Collection {
+	return docstore.NewCollection(newCollection(mcoll, idField, nil))
+}
+
+// OpenCollectionFunc opens a MongoDB collection for use with Docstore.
```
Comment (vangent) [inline internal/docstore/mongodocstore/mongo.go]: Similar to other PR, maybe `OpenCollectionIDFieldFunc`?

File: internal/docstore/mongodocstore/mongo_test.go [hunk 6]
```diff
@@ -42,7 +42,7 @@ type harness struct {
 }
 
 func (h *harness) MakeCollection(ctx context.Context) (driver.Collection, error) {
-	coll := newCollection(h.db.Collection(collectionName), Options{IDField: "_id"})
+	coll := newCollection(h.db.Collection(collectionName), "_id", nil)
```
Comment (vangent) [inline internal/docstore/mongodocstore/mongo_test.go]: Or just "", right?

Comment (jba) [reply internal/docstore/mongodocstore/mongo_test.go]: Done.

**Pull request merged**